### PR TITLE
chore: add component generator templates with plop

### PR DESCRIPTION
**Pull request by anniehu4**
### Summary:
Generators are super cool and thanks to https://github.com/bradfrost/czi-vanilla-storybook/blob/master/plopfile.js, it's pretty easy for us to set up 🤓 

Here's what this outputs:
Screen Shot 2021-02-22 at 3 46 09 PM

- Templates in `plop-templates/**/*.hbs`
- Generated files (for review only) in `src/PlopTest/*`

### Test Plan:
I included the files from running:
```
plop
component name: "PlopTest"
```
for review (so CI failures expected)! will remove all that generated stuff before landing

### CODE REVIEW

**Comment by anniehu4** [comment]
Note: this is very not urgent, I was just intrigued

**Review by ahuth — APPROVED** [review]
Nice, LGTM 🤘

File: plop-templates/Component/Component.tsx.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+function {{pascalCase name}}({ children, ...rest }: Props) {
+  return (
+    <{{pascalCase name}}Component {...rest}>
+      {children}
+    </{{pascalCase name}}Component>
+  );
+}
+
+export default {{pascalCase name}};
```
Comment (dcwither) [inline plop-templates/Component/Component.tsx.hbs:23]: I've actually been working on something similar for a personal project template. One thing I added was defaultProps with an example along with the matching types on Props.

```
{{pascalCase name}}.defaultProps = {
  name: '{{pascalCase name}}'
}
```

File: plop-templates/Component/Component.tsx.hbs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import tw, { styled } from "twin.macro";
+
+type Props = {
+  /**
+  * Child node(s) that can be nested inside component
+  */
+  children: ReactNode;
+};
+
+interface Base{{pascalCase name}}Props {}
```
Comment (dcwither) [inline plop-templates/Component/Component.tsx.hbs:11]: How does this play into Props, should Props join it in?

**Comment by anniehu4** [comment]
done in https://github.com/chanzuckerberg/edu-design-system/pull/366

**Pull request closed without merging**